### PR TITLE
Shiqi

### PR DESCRIPTION
**Pull request by shiqio3o**
Update Folder dao, logged in view, my folder view, opened folder view, and combine open folder and opened folder into one use case

the part related to "create folder" and "open recipe" still need to be checked

### CODE REVIEW

File: src/interface_adapter/my_folder/MyFolderPresenter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        @Override
+        public void prepareSuccessView(MyFolderOutputData response) {
+            // On success, switch to the logged in view.
+
+            OpenedFolderState openedFolderState = openedFolderViewModel.getState();
+            openedFolderState.setFoldername(response.getFoldername());
+            this.openedFolderViewModel.setState(openedFolderState);
+            this.openedFolderViewModel.firePropertyChanged();
```
Comment (yuxizzz) [inline src/interface_adapter/my_folder/MyFolderPresenter.java:34]: why did you use openedFolderView? shouldn't it be myFolderView for this use case?

File: src/use_case/my_folder/MyFolderInputData.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import entity.recipe.Recipe;
+
+import java.util.HashMap;
+
+public class MyFolderInputData {
+
+    final private String foldername;
+
+    public MyFolderInputData(String foldername) {
```
Comment (yuxizzz) [inline src/use_case/my_folder/MyFolderInputData.java:11]: Ok, just a clarification. MyFolder is something in our login view that we click, we get it to another view with a list of folders this user have. So why do you use foldername as an input.

File: src/use_case/my_folder/MyFolderInteractor.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        this.folderDataAccessObject = folderDataAccessObject;
+        this.myFolderPresenter = myFolderPresenter;
+    }
+
+    public void execute(MyFolderInputData myFolderInputData) {
+        String foldername = myFolderInputData.getFoldername();
+        if (!folderDataAccessObject.existsByName(foldername)) {
+            myFolderPresenter.prepareFailView(foldername + ": Folder does not exist.");
+        }else {
+             Folder folder = folderDataAccessObject.get(myFolderInputData.getFoldername());
```
Comment (yuxizzz) [inline src/use_case/my_folder/MyFolderInteractor.java:25]: yeah, this is it, this is not exactly the same as the openfolder usecase that you implemented right, they are 2 different things

**Review by yuxizzz — COMMENTED** [review]
this should be dfferent from openfolder my dear.

File: src/view/MyFolderView.java [hunk 4]
```diff
@@ -40,7 +67,17 @@ public MyFolderView(MyFolderViewModel myFolderViewModel, LogoutViewModel logoutV
         username = new JLabel();
 
         JPanel buttons = new JPanel();
-        logOut = new JButton(myFolderViewModel.LOGOUT_BUTTON_LABEL);
+        JButton logOut = new JButton(myFolderViewModel.LOGOUT_BUTTON_LABEL);
+        JButton openFolder = new JButton(myFolderViewModel.OPEN_BUTTON_LABEL);
```
Comment (yuxizzz) [inline src/view/MyFolderView.java:78]: so does it mean this user are only allowed to have 1 folder?

File: src/view/LoggedInView.java [hunk 5]
```diff
@@ -61,22 +74,56 @@ public LoggedInView(LoggedInViewModel loggedInViewModel,
...
         buttons.add(search);
+        buttons.add(myFolder);
+        buttons.add(logOut);
+
 
         search.addActionListener(
                 new ActionListener() {
                     @Override
                     public void actionPerformed(ActionEvent evt) {
                         if (evt.getSource().equals(search)){
```
Comment (yuxizzz) [inline src/view/LoggedInView.java:86]: this is great

File: src/interface_adapter/my_folder/MyFolderPresenter.java (hunk 1 above)
Comment (shiqio3o) [reply src/interface_adapter/my_folder/MyFolderPresenter.java:34]: this means we transfer it to the opened folder view which is the next view of my folder view

File: src/use_case/my_folder/MyFolderInputData.java (hunk 2 above)
Comment (shiqio3o) [reply src/use_case/my_folder/MyFolderInputData.java:11]: yes, i think it should input username

File: src/view/MyFolderView.java (hunk 4 above)
Comment (shiqio3o) [reply src/view/MyFolderView.java:78]: I have a for-loop below to address all folders

**Review by yuxizzz — APPROVED** [review]

**Pull request merged**